### PR TITLE
Notification events for add/delete

### PR DESCRIPTION
**Pull request by mloock**
Hi,

This PR is intented to bring new events regarding the notification feature.
This adds the pre/post add/update/delete events.
For example, this could be useful to send an email for a new notification.

### CODE REVIEW

File: lib/Event/Model/NotificationEvent.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+     */
+    public function getNotification()
+    {
+        return $this->notification;
+    }
+
+    /**
+     * @param Notification $notification
+     */
+    public function setObject($notification)
```
[the remark this answers is not in the document]
Comment (mloock) [reply lib/Event/Model/NotificationEvent.php]: You're right! :)

**Comment by brusch** [comment]
I'd prefer a bit simpler event structure, with just `pre/post save` and `pre/post delete` just as other simple entities do, see https://github.com/pimcore/pimcore/blob/bf4799782bb6d10ebe2adab85d9eb555ce0b84f9/lib/Event/RedirectEvents.php#L17 for example. 

Can you please update the PR accordingly? 
Thanks a lot!

**Comment by mloock** [comment]
> I'd prefer a bit simpler event structure, with just `pre/post save` and `pre/post delete` just as other simple entities do, see
> 
> https://github.com/pimcore/pimcore/blob/bf4799782bb6d10ebe2adab85d9eb555ce0b84f9/lib/Event/RedirectEvents.php#L17
> 
> for example.
> Can you please update the PR accordingly?
> Thanks a lot!

Hi @brusch !
Of course, I'll PR this in a few minutes ;)
The update events would have been useful for the first read action, because we update the actual notification object in this case.
But no problem, I amend that right now!

Thanks for the review

File: lib/Event/NotificationEvents.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+namespace Pimcore\Event;
+
+final class NotificationEvents
+{
+    /**
+     * @Event("Pimcore\Event\Model\NotificationEvent")
+     *
+     * @var string
+     */
+    const PRE_ADD = 'pimcore.notification.preAdd';
```
Comment (brusch) [inline lib/Event/NotificationEvents.php:24]: This should be `PRE_SAVE`

File: lib/Event/NotificationEvents.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    /**
+     * @Event("Pimcore\Event\Model\NotificationEvent")
+     *
+     * @var string
+     */
+    const POST_ADD = 'pimcore.notification.postAdd';
```
Comment (brusch) [inline lib/Event/NotificationEvents.php:31]: This should be `POST_SAVE`

**Comment by brusch** [comment]
That looks good now! Thanks a lot!

**Pull request merged**